### PR TITLE
metrics: add data flow metrics

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add data flow metrics.

image

Also, this PR adds chechkpoint derivative and TiKV traffics

image
image

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

Cc #1606 

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/entry/mounter.go [hunk 1]
```diff
@@ -138,6 +139,7 @@ func (m *mounterImpl) codecWorker(ctx context.Context, index int) error {
 		pEvent.RawKV.OldValue = nil
 		pEvent.PrepareFinished()
 		metricMountDuration.Observe(time.Since(startTime).Seconds())
+		metricTotalRows.Add(1)
```
Comment (3AceShowHand) [inline cdc/entry/mounter.go:142]: how about using `Inc()`, and shall we delete this metrics when mounter close.

File: cdc/processor/pipeline/sink.go [hunk 2]
```diff
@@ -150,6 +151,7 @@ func (n *sinkNode) flushSink(ctx pipeline.NodeContext, resolvedTs model.Ts) (err
 
 func (n *sinkNode) emitEvent(ctx pipeline.NodeContext, event *model.PolymorphicEvent) error {
 	if event == nil || event.Row == nil {
+		log.Warn("skip emit empty rows", zap.Reflect("event", event))
```
Comment (3AceShowHand) [inline cdc/processor/pipeline/sink.go:154]: `zap.Any` is preferred than `zap.Reflect`

File: cdc/sink/manager.go [hunk 3]
```diff
@@ -43,16 +44,22 @@ type Manager struct {
 	flushing int64
 
 	drawbackChan chan drawbackMsg
+
+	metricsTableSinkTotalRows prometheus.Counter
```
Comment (3AceShowHand) [inline cdc/sink/manager.go:48]: shall we delete this when closing the sink manager?

File: cdc/sink/manager.go [hunk 4]
```diff
@@ -237,6 +245,7 @@ func (b *bufferSink) run(ctx context.Context, errCh chan error) {
 	metricFlushDuration := flushRowChangedDuration.WithLabelValues(advertiseAddr, changefeedID, "Flush")
 	metricEmitRowDuration := flushRowChangedDuration.WithLabelValues(advertiseAddr, changefeedID, "EmitRow")
 	metricBufferSize := bufferChanSizeGauge.WithLabelValues(advertiseAddr, changefeedID)
+	metricTotalRows := bufferSinkTotalRowsCountCounter.WithLabelValues(advertiseAddr, changefeedID)
```
Comment (3AceShowHand) [inline cdc/sink/manager.go:248]: ditto

File: metrics/grafana/ticdc.json [hunk 5]
```diff
@@ -3618,27 +3660,922 @@
...
+          "paceLength": 10,
+          "percentage": false,
+          "pointradius": 2,
+          "points": false,
+          "renderer": "flot",
+          "seriesOverrides": [
+            {
+              "alias": "/.*-rpc/",
+              "yaxis": 2
+            }
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:4475]: Is this useless

File: metrics/grafana/ticdc.json [hunk 6]
```diff
@@ -4080,32 +4993,44 @@
...
           "percentage": false,
           "pointradius": 2,
           "points": false,
           "renderer": "flot",
-          "seriesOverrides": [],
+          "seriesOverrides": [
+            {
+              "alias": "/.*processor.*/",
+              "yaxis": 2
+            }
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:5005]: ditto

**Comment by amyangfei** [comment]
CI failed

File: cdc/sink/metrics.go [hunk 7]
```diff
@@ -69,14 +69,14 @@ var (
...
-			Help:      "totla count of rows",
+			Help:      "The total count of rows that are processed by sink",
 		}, []string{"capture", "changefeed"})
 	totalFlushedRowsCountGauge = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Namespace: "ticdc",
 			Subsystem: "sink",
 			Name:      "total_flushed_rows_count",
-			Help:      "totla count of flushed rows",
+			Help:      "total count of flushed rows",
```
Comment (overvenus) [inline cdc/sink/metrics.go:79]:
```suggestion
			Help:      "The total count of rows that are flushed by sink",
```

**Review by 3AceShowHand — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**